player.go

```diff
@@ -0,0 +1,493 @@
+//
+// Only expects PlayOptions.DeviceID, all other options will be ignored
+func (c *Client) NextOpt(opt *PlayOptions) error {
+	spotifyURL := baseAddress + "me/player/next"
+
+	if opt != nil {
+		v := url.Values{}
+		if opt.DeviceID != nil {
+			v.Set("device_id", *opt.DeviceID)
+		}
+		if params := v.Encode(); params != "" {
+			spotifyURL += "?" + params
+		}
+	}
+	req, err := http.NewRequest(http.MethodPost, spotifyURL, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusNoContent {
+		return decodeError(resp.Body)
+	}
+	return nil
+}
+
+// Previous skips to the Previous track in the user's queue in the user's
+// currently active device. This call requires authorization.
+//
+// Requires the ScopeUserModifyPlaybackState in order to modify the player state
+func (c *Client) Previous() error {
+	return c.NextOpt(nil)
+}
+
+// PreviousOpt Like Previous but with more options
+//
+// Only expects PlayOptions.DeviceID, all other options will be ignored
+func (c *Client) PreviousOpt(opt *PlayOptions) error {
+	spotifyURL := baseAddress + "me/player/previous"
+
+	if opt != nil {
+		v := url.Values{}
+		if opt.DeviceID != nil {
+			v.Set("device_id", *opt.DeviceID)
+		}
+		if params := v.Encode(); params != "" {
+			spotifyURL += "?" + params
+		}
+	}
+	req, err := http.NewRequest(http.MethodPost, spotifyURL, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusNoContent {
+		return decodeError(resp.Body)
+	}
+	return nil
+}
+
+// Seek to the given position in the user’s currently playing track.
+// This call requires authorization.
+//
+// The position in milliseconds to seek to. Must be a positive number.
+// Passing in a position that is greater than the length of the track
+// will cause the player to start playing the next song.
+//
+// Requires the ScopeUserModifyPlaybackState in order to modify the player state
+func (c *Client) Seek(position int) error {
+	return c.SeekOpt(position, nil)
+}
+
+// SeekOpt Like Seek but with more options
+//
+// Only expects PlayOptions.DeviceID, all other options will be ignored
+func (c *Client) SeekOpt(position int, opt *PlayOptions) error {
+	return c.playerFuncWithOpt(
+		"me/player/seek",
+		url.Values{
+			"position_ms": []string{strconv.FormatInt(int64(position), 10)},
+		},
+		opt,
+	)
+}
+
+// Repeat Set the repeat mode for the user's playback.
+// This call requires authorization.
+//
+// Options are repeat-track, repeat-context, and off.
+//
+// Requires the ScopeUserModifyPlaybackState in order to modify the player state
+func (c *Client) Repeat(state string) error {
+	return c.RepeatOpt(state, nil)
+}
+
+// RepeatOpt Like Repeat but with more options
+//
+// Only expects PlayOptions.DeviceID, all other options will be ignored
+func (c *Client) RepeatOpt(state string, opt *PlayOptions) error {
```

Comment:
I was going for consistency (Ie: things suffixed 'opt' take opts...) but if you like I can do that.